packages/x-data-grid-pro/src/components/GridPinnedRows.tsx

```diff
@@ -20,18 +19,17 @@ export function GridPinnedRows({ position, virtualScroller }: GridPinnedRowsProp
   const classes = useUtilityClasses();
   const apiRef = useGridPrivateApiContext();
 
-  const renderContext = useGridSelector(apiRef, gridRenderContextSelector);
   const pinnedRowsData = useGridSelector(apiRef, gridPinnedRowsSelector);
   const rows = pinnedRowsData[position];
 
   const pinnedRenderContext = React.useMemo(
     () => ({
       firstRowIndex: 0,
       lastRowIndex: rows.length,
-      firstColumnIndex: renderContext.firstColumnIndex,
-      lastColumnIndex: renderContext.lastColumnIndex,
+      firstColumnIndex: undefined,
+      lastColumnIndex: undefined,
```

Comment:
If this value isn't used, can we use `-1` as an empty value marker instead? It will [avoid changing the shape](https://romgrk.com/posts/optimizing-javascript#2-avoid-different-shapes) of those objects.